contracts/colonyNetwork/IColonyNetwork.sol

```diff
@@ -285,6 +295,45 @@ contract IColonyNetwork is ColonyNetworkDataTypes, IRecovery {
   /// @return miningResolverAddress The address of the mining cycle resolver currently used by new instances
   function getMiningResolver() public view returns (address miningResolverAddress);
 
+  /// @notice Add a new extension resolver to the Extensions repository.
+  /// @dev Can only be called by the MetaColony.
+  /// @dev The extension version is queried from the resolver itself.
+  /// @dev The roles array can be set only for version == 1 (must be empty otherwise).
+  /// @param extensionId keccak256 hash of the extension name, used as an indentifier
+  /// @param resolver The deployed resolver containing the extension contract logic
+  function addExtension(bytes32 extensionId, address resolver) public;
+
+  /// @notice Install an extension in a colony. Can only be called by a Colony.
+  /// @param extensionId keccak256 hash of the extension name, used as an indentifier
+  /// @param version Version of the extension to install
+  function installExtension(bytes32 extensionId, uint256 version) public;
+
+  /// @notice Upgrade an extension in a colony. Can only be called by a Colony.
+  /// @param extensionId keccak256 hash of the extension name, used as an indentifier
+  /// @param newVersion Version of the extension to upgrade to (must be one greater than current)
+  function upgradeExtension(bytes32 extensionId, uint256 newVersion) public;
+
+  /// @notice Deprecate an extension in a colony. Can only be called by a Colony.
```

Comment:
Or undeprecate. I note the ability to undeprecate isn't really reflected in the name of the function, but I'm not really sure what we'd rename it to to improve that.

Reply:
We could call it "setExtensionDeprecation" but I think that passing in a boolean is sufficiently clear.